lib/algolia/client.rb

```diff
@@ -439,7 +439,7 @@ def delete(uri, type = :write, request_options = {})
     def thread_local_hosts(read)
       thread_hosts_key = read ? "algolia_search_hosts_#{application_id}" : "algolia_hosts_#{application_id}"
       Thread.current[thread_hosts_key] ||= (read ? search_hosts : hosts).each_with_index.map do |host, i|
-        client = HTTPClient.new
+        client = HTTPClient.new(ENV["HTTP_PROXY"])
```

Comment:
This shouldn't be needed because the `httpclient` library already read this variables:
 - documented in https://github.com/nahi/httpclient/blob/master/lib/httpclient.rb#L46-L47
 - loaded from https://github.com/nahi/httpclient/blob/master/lib/httpclient.rb#L1065-L1077
 - called from https://github.com/nahi/httpclient/blob/master/lib/httpclient.rb#L449

Any chance you could tell us what happens in this `load_environment` method in your case @vinc ? Maybe you have a `REQUEST_METHOD` variable forcing the code to look for `cgi_http_proxy` ?

Reply:
Indeed yesterday our fork seemed to fix the issue we had on our production server, but we switched back to the original gem after reading your comment and everything is still working.

Therefor the issue we were having seems to have been totally unrelated!

So we can close this PR, and I apologize for your time spent on this :3

Reply:
No problem @vinc thanks for contributing!